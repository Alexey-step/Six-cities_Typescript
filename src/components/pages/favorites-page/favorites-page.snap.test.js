import React from 'react';
import {render} from '@testing-library/react';
import FavoritesPage from './favorites-page';
import * as redux from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import {offers, authInfo} from '../../../mocks/tests-mocks';

const mockStore = configureStore({});
describe(`Test FavoritesPage component`, () => {
  const history = createMemoryHistory();
  it(`FavoritesPage component should render correctly`, () => {
    const store = mockStore({
      MAIN: {isError: false},
      USER: {authInfo: authInfo.adapted},
      FAVORITE: {favorites: offers.adapted}
    });
    store.dispatch = () => {};
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <FavoritesPage />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
  it(`FavoritesPage component should render correctly when favorites empty`, () => {
    const store = mockStore({
      MAIN: {isError: false},
      USER: {authInfo: authInfo.adapted},
      FAVORITE: {favorites: []}
    });
    store.dispatch = () => {};
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <FavoritesPage />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
