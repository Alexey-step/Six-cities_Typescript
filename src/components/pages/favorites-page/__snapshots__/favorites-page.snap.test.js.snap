// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FavoritesPage component FavoritesPage component should render correctly 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg);"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    mail@gmail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <button
                  class="header__nav-link header__nav-link--profile header__nav-link--logout"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Logout
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites"
        >
          <h1
            class="favorites__title"
          >
            Saved listing
          </h1>
          <ul
            class="favorites__list"
          >
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Paris
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places places__list "
              >
                <article
                  class="favorites__card place-card"
                  data-testid="mouse"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                    data-testid="scroll"
                  >
                    <a
                      href="/offer/1"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          161
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button button false"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            href="#icon-bookmark"
                            xlink="true"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 40%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/1"
                      >
                        Tile House
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      house
                    </p>
                  </div>
                </article>
              </div>
            </li>
            
            
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Amsterdam
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places places__list "
              >
                <article
                  class="favorites__card place-card"
                  data-testid="mouse"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                    data-testid="scroll"
                  >
                    <a
                      href="/offer/2"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          161
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button button false"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            href="#icon-bookmark"
                            xlink="true"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 80%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/2"
                      >
                        Tile House
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      house
                    </p>
                  </div>
                </article>
              </div>
            </li>
            
            
          </ul>
        </section>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;

exports[`Test FavoritesPage component FavoritesPage component should render correctly when favorites empty 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg);"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    mail@gmail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <button
                  class="header__nav-link header__nav-link--profile header__nav-link--logout"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Logout
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites page__main--favorites-empty"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites favorites--empty"
        >
          <h1
            class="visually-hidden"
          >
            Favorites (empty)
          </h1>
          <div
            class="favorites__status-wrapper"
          >
            <b
              class="favorites__status"
            >
              Nothing yet saved.
            </b>
            <p
              class="favorites__status-description"
            >
              Save properties to narrow down search or plan your future trips.
            </p>
          </div>
        </section>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;
